Outlookv2

rbreesems opened the pull request:
This pull request does the following:

  1. Adds a new handler (Handlers/Outlookv2.cs) that does email actions based on probabilities.  See the sample timeline for this. This is the email handler that we are using for our email traffic gen.
 
  2. Removed AutoIt usage from Pidgin as it causes Outlook problems and replaced all usages with Windows Sendkeys, as we saw the Outlook thread would immediately bomb  once Pidgin started. However, running the Pidgin handler with Office (2013, 2019, 2021) for  a few hours still causes Outlook to lock up, and we traced it to the use of Sendkeys.    For some reason, on VMware VMs, this combination causes instability. So, Pidgin is fine, as long as you don't run it with the Outlook handler.

 3. Removed all usages of AutoIt and Windows Sendkeys from the Notepad handler because of the above clashes with Outlook. This also meant removing the PDF generation capability that was in there. Now the Notepad handler just generates random text files.

This pull request DOES NOT include the project files as our project files have diverged.  To include this pull request, the Handlers/Outlookv2.cs  file and the Sample Timelines/Outlookv2.json will have to be added to your GHOSTS client project file.

This will be the last pull request for a new feature for a while. We are happy with GHOSTS capabilities at the moment and are concentrating on deployment, testing.  We have deployed GHOSTS in an active event, and are planning on including it all future events.

sei-dupdyke commented:
This is pulled and published in the latest binaries (v7.0.111), thank you, and thanks for the update on use, always appreciated.

Dustin
________________________________
From: Bob Reese ***@***.***>
Sent: Friday, March 10, 2023 10:14 AM
To: cmu-sei/GHOSTS ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [cmu-sei/GHOSTS] Outlookv2 (PR #189)


This pull request does the following:

  1.  Adds a new handler (Handlers/Outlookv2.cs) that does email actions based on probabilities. See the sample timeline for this. This is the email handler that we are using for our email traffic gen.

  2.  Removed AutoIt usage from Pidgin as it causes Outlook problems and replaced all usages with Windows Sendkeys, as we saw the Outlook thread would immediately bomb once Pidgin started. However, running the Pidgin handler with Office (2013, 2019, 2021) for a few hours still causes Outlook to lock up, and we traced it to the use of Sendkeys. For some reason, on VMware VMs, this combination causes instability. So, Pidgin is fine, as long as you don't run it with the Outlook handler.

  3.  Removed all usages of AutoIt and Windows Sendkeys from the Notepad handler because of the above clashes with Outlook. This also meant removing the PDF generation capability that was in there. Now the Notepad handler just generates random text files.

This pull request DOES NOT include the project files as our project files have diverged. To include this pull request, the Handlers/Outlookv2.cs file and the Sample Timelines/Outlookv2.json will have to be added to your GHOSTS client project file.

This will be the last pull request for a new feature for a while. We are happy with GHOSTS capabilities at the moment and are concentrating on deployment, testing. We have deployed GHOSTS in an active event, and are planning on including it all future events.

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/cmu-sei/GHOSTS/pull/189

Commit Summary

  *   a697346<https://github.com/cmu-sei/GHOSTS/pull/189/commits/a69734602f921693546c5295837412b972092220> Initial push for outlookv2 handler.
  *   e728a0d<https://github.com/cmu-sei/GHOSTS/pull/189/commits/e728a0df0be946a8e64bcd57e1aa3c16a34a719b> Outlookv2 update.
  *   d61c464<https://github.com/cmu-sei/GHOSTS/pull/189/commits/d61c464cf8ed3aafd5c60378dab02aae634b8552> Changed deleted to always delete last item as suggested in API.
  *   86bbd2f<https://github.com/cmu-sei/GHOSTS/pull/189/commits/86bbd2f8defde13c38ae4969ce99953799a8a830> Adjusted how deletion is done.
  *   4533109<https://github.com/cmu-sei/GHOSTS/pull/189/commits/45331095a835e92cca64765b7865e98883c76586> Removed AutoIt from Pidgin handler.

File Changes

(9 files<https://github.com/cmu-sei/GHOSTS/pull/189/files>)

  *   M src/Ghosts.Client/Handlers/Notepad.cs<https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-f2ed2278da39f236500280585777260d133aa87e16ccd566f3a16af6f7763213> (148)
  *   A src/Ghosts.Client/Handlers/Outlookv2.cs<https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-5e3f41f1d5fa7d4e2a8f22a00280f767a7a0a064234d781a8c956884ebdac6a0> (1098)
  *   M src/Ghosts.Client/Handlers/Pidgin.cs<https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-ca99ab5c9d062b5ed8a7fd275dac9cca07df6c42381ae9b8df93c99c1e6e8862> (32)
  *   M src/Ghosts.Client/Infrastructure/CronScheduling.cs<https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-79c6e40fc237959fdd6a8632929b66d48c8857f5a8750690e3ff28f2aadf3360> (2)
  *   A src/Ghosts.Client/Sample Timelines/Outlookv2.json<https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-0485ea32a2f216712597326e4f0001a1e1e6345bbab4ff2cddd7c6a7e0d8564a> (63)
  *   M src/Ghosts.Client/TimelineManager/Orchestrator.cs<https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-eb635d323ec7eb0ddc8023ce23c49bed6376696e253d302dc0a362cfe3203757> (8)
  *   M src/Ghosts.Client/config/application.json<https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-2b596969cdf2fbc27237680faa7e772f1eed3f54307bae423be066c180899bcd> (1)
  *   M src/Ghosts.Domain/Code/ClientConfiguration.cs<https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-215db36e9a01bb7bbef8ea6fd56de7c64361a485ee1081260145476f36d6daab> (1)
  *   M src/Ghosts.Domain/Messages/Timeline.cs<https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-77f38a578271ba868538f9c6beacdd851a3beae228892fa52ba20bb9c5a5755b> (1)

Patch Links:

  *   https://github.com/cmu-sei/GHOSTS/pull/189.patch
  *   https://github.com/cmu-sei/GHOSTS/pull/189.diff

—
Reply to this email directly, view it on GitHub<https://github.com/cmu-sei/GHOSTS/pull/189>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AKLOR4A6VIUBRCM2PBZ7SRTW3NAMTANCNFSM6AAAAAAVWSQRRE>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

rbreesems commented:
Thanks for pulling this!

I will sync up with Master going forward, will just be bug-fixes for a
while as we extensively test.

Bob


On Fri, Mar 10, 2023 at 2:40 PM Dustin Updyke ***@***.***>
wrote:

> This is pulled and published in the latest binaries (v7.0.111), thank you,
> and thanks for the update on use, always appreciated.
>
> Dustin
> ________________________________
> From: Bob Reese ***@***.***>
> Sent: Friday, March 10, 2023 10:14 AM
> To: cmu-sei/GHOSTS ***@***.***>
> Cc: Subscribed ***@***.***>
> Subject: [cmu-sei/GHOSTS] Outlookv2 (PR #189)
>
>
> This pull request does the following:
>
> 1. Adds a new handler (Handlers/Outlookv2.cs) that does email actions
> based on probabilities. See the sample timeline for this. This is the email
> handler that we are using for our email traffic gen.
>
> 2. Removed AutoIt usage from Pidgin as it causes Outlook problems and
> replaced all usages with Windows Sendkeys, as we saw the Outlook thread
> would immediately bomb once Pidgin started. However, running the Pidgin
> handler with Office (2013, 2019, 2021) for a few hours still causes Outlook
> to lock up, and we traced it to the use of Sendkeys. For some reason, on
> VMware VMs, this combination causes instability. So, Pidgin is fine, as
> long as you don't run it with the Outlook handler.
>
> 3. Removed all usages of AutoIt and Windows Sendkeys from the Notepad
> handler because of the above clashes with Outlook. This also meant removing
> the PDF generation capability that was in there. Now the Notepad handler
> just generates random text files.
>
> This pull request DOES NOT include the project files as our project files
> have diverged. To include this pull request, the Handlers/Outlookv2.cs file
> and the Sample Timelines/Outlookv2.json will have to be added to your
> GHOSTS client project file.
>
> This will be the last pull request for a new feature for a while. We are
> happy with GHOSTS capabilities at the moment and are concentrating on
> deployment, testing. We have deployed GHOSTS in an active event, and are
> planning on including it all future events.
>
> ________________________________
> You can view, comment on, or merge this pull request online at:
>
> https://github.com/cmu-sei/GHOSTS/pull/189
>
> Commit Summary
>
> * a697346<
> https://github.com/cmu-sei/GHOSTS/pull/189/commits/a69734602f921693546c5295837412b972092220>
> Initial push for outlookv2 handler.
> * e728a0d<
> https://github.com/cmu-sei/GHOSTS/pull/189/commits/e728a0df0be946a8e64bcd57e1aa3c16a34a719b>
> Outlookv2 update.
> * d61c464<
> https://github.com/cmu-sei/GHOSTS/pull/189/commits/d61c464cf8ed3aafd5c60378dab02aae634b8552>
> Changed deleted to always delete last item as suggested in API.
> * 86bbd2f<
> https://github.com/cmu-sei/GHOSTS/pull/189/commits/86bbd2f8defde13c38ae4969ce99953799a8a830>
> Adjusted how deletion is done.
> * 4533109<
> https://github.com/cmu-sei/GHOSTS/pull/189/commits/45331095a835e92cca64765b7865e98883c76586>
> Removed AutoIt from Pidgin handler.
>
> File Changes
>
> (9 files<https://github.com/cmu-sei/GHOSTS/pull/189/files>)
>
> * M src/Ghosts.Client/Handlers/Notepad.cs<
> https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-f2ed2278da39f236500280585777260d133aa87e16ccd566f3a16af6f7763213>
> (148)
> * A src/Ghosts.Client/Handlers/Outlookv2.cs<
> https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-5e3f41f1d5fa7d4e2a8f22a00280f767a7a0a064234d781a8c956884ebdac6a0>
> (1098)
> * M src/Ghosts.Client/Handlers/Pidgin.cs<
> https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-ca99ab5c9d062b5ed8a7fd275dac9cca07df6c42381ae9b8df93c99c1e6e8862>
> (32)
> * M src/Ghosts.Client/Infrastructure/CronScheduling.cs<
> https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-79c6e40fc237959fdd6a8632929b66d48c8857f5a8750690e3ff28f2aadf3360>
> (2)
> * A src/Ghosts.Client/Sample Timelines/Outlookv2.json<
> https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-0485ea32a2f216712597326e4f0001a1e1e6345bbab4ff2cddd7c6a7e0d8564a>
> (63)
> * M src/Ghosts.Client/TimelineManager/Orchestrator.cs<
> https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-eb635d323ec7eb0ddc8023ce23c49bed6376696e253d302dc0a362cfe3203757>
> (8)
> * M src/Ghosts.Client/config/application.json<
> https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-2b596969cdf2fbc27237680faa7e772f1eed3f54307bae423be066c180899bcd>
> (1)
> * M src/Ghosts.Domain/Code/ClientConfiguration.cs<
> https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-215db36e9a01bb7bbef8ea6fd56de7c64361a485ee1081260145476f36d6daab>
> (1)
> * M src/Ghosts.Domain/Messages/Timeline.cs<
> https://github.com/cmu-sei/GHOSTS/pull/189/files#diff-77f38a578271ba868538f9c6beacdd851a3beae228892fa52ba20bb9c5a5755b>
> (1)
>
> Patch Links:
>
> * https://github.com/cmu-sei/GHOSTS/pull/189.patch
> * https://github.com/cmu-sei/GHOSTS/pull/189.diff
>
> —
> Reply to this email directly, view it on GitHub<
> https://github.com/cmu-sei/GHOSTS/pull/189>, or unsubscribe<
> https://github.com/notifications/unsubscribe-auth/AKLOR4A6VIUBRCM2PBZ7SRTW3NAMTANCNFSM6AAAAAAVWSQRRE
> >.
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cmu-sei/GHOSTS/pull/189#issuecomment-1464434408>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABGBHPOHEVJN4SIAL3J75ZDW3OGUBANCNFSM6AAAAAAVWSQRRE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
